contrib/src/main/java/com/datatorrent/contrib/parser/RegexSplitter.java

```diff
@@ -0,0 +1,129 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.datatorrent.contrib.parser;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.util.Date;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import javax.validation.constraints.NotNull;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.apache.commons.beanutils.BeanUtils;
+import org.apache.commons.beanutils.ConversionException;
+import org.apache.commons.beanutils.ConvertUtils;
+import org.apache.commons.beanutils.converters.DateConverter;
+import org.apache.commons.beanutils.converters.DateTimeConverter;
+import org.apache.commons.lang.StringUtils;
+
+import com.datatorrent.api.Context;
+import com.datatorrent.lib.util.KeyValPair;
+
+/**
+ * Operator parsers the tuple based on regex pattern and populates POJO matching the user defined schema <br>
+ * This operator expects the upstream operator to send every line in the file as a byte array.
+ * splitRegexPattern contains the regex pattern of lines in the file <br>
+ * Schema is specified in a json format as per {@link DelimitedSchema} that
+ * contains field information and constraints for each field.<br>
+ * Schema field names should match with the POJO variable names<br>
+ *Assumption is that each field in the delimited data should map to a simple
+ * java type.<br>
+ * <br>
+ * <b>Properties</b> <br>
+ * <b>splitRegexPattern</b>:Regex pattern as a string<br>
+ *
+ * @displayName RegexSplitter
+ * @category Parsers
+ * @tags pojo parser regex logs server
+ * @since 3.6.0
+ */
+public class RegexSplitter extends CsvParser
```

Comment:
Any specific reason to extend from CsvParser ?  I feel RegexSplitter is not a "special case" of Csv Parser and hence it should extend from Parser instead of CsvParser.